### DISCUSSION TITLE
Bus::fake errors on laravel 10 update

### DISCUSSION THREAD

**Discussion by danielsballes** [laravel/framework#51622] [Q&A]
Hi, after upgrading a Laravel project from version 9 to 10 I'm seeing a bunch of errors in the tests where we're using the `Bus::fake` facade, for example, I have this one:

```php
class DeleteOffersForCampaignTest extends TestCase
{
    use DatabaseTransactions, WithFaker;

    private Campaign $campaign;

    protected function setUp(): void
    {
        parent::setUp();
        Bus::fake([SyncCampaignToAppsJob::class, UpdateOffersForCampaignJob::class]);

        $this->campaign = Campaign::factory()->active()->create();
    }

    public function test_updating_campaign_status_to_not_active_dispatches_delete_offers_for_campaign_job()
    {
        Bus::fake(DeleteOffersForCampaign::class);

        $campaign = Campaign::factory()->active()->create();

        $campaign->update([
            'campaign_status' => 'paused',
        ]);

        Bus::assertDispatched(function (DeleteOffersForCampaign $job) use ($campaign) {
            return $job->campaign->id === $campaign->id;
        });
    }
}
``` 

This is the error:

> Cannot assign requested address [tcp://localhost:6379]
 at vendor/predis/predis/src/Connection/AbstractConnection.php:155
    151▕      */
    152▕     protected function onConnectionError($message, $code = null)
    153▕     {
    154▕         CommunicationException::handle(
  ➜ 155▕             new ConnectionException($this, static::createExceptionMessage($message), $code)
    156▕         );
    157▕     }
    158▕ 
    159▕     /**
+25 vendor frames 
  26  app/Listeners/SyncCampaignToAppsAndUpdateOffersForCampaign.php:34
      +8 vendor frames 
  35  tests/Feature/Jobs/DeleteOffersForCampaignTest.php:37

I have already found a solution for this error, but I don't have any idea why in the previous version it was working correctly, I hope anyone could give me an answer, this is driving me crazy.

**Comment by willpower232** [comment]
Hello from the future

https://github.com/laravel/framework/pull/46257

I reckon this is to blame, makes sense but annoying if you're trying to fake jobs across every test AND in specific tests
